packages/oid4vci-holder/src/agent/OID4VCIHolder.ts

```diff
@@ -1016,6 +1008,34 @@ export class OID4VCIHolder implements IAgentPlugin {
     logger.log(`Notification to ${notificationEndpoint} has been dispatched`)
   }
 
+  private async oid4vciGetFederationTrust(args: GetFederationTrustArgs, context: RequiredContext): Promise<boolean> {
+    const { requestData, serverMetadata, trustAnchors } = args
+
+    if (trustAnchors.length === 0) {
+      return Promise.reject(Error('No trust anchors found'))
+    }
+
+    if (!requestData?.uri) {
+      return Promise.reject(Error('Missing request URI in context'))
+    }
+
+    if (!serverMetadata) {
+      return Promise.reject(Error('Missing serverMetadata in context'))
+    }
+
+    const url = new URL(requestData?.uri)
+    const params = new URLSearchParams(url.search)
+    const openidFederation = params.get('openid_federation')
+
+    const entityIdentifier = openidFederation ?? serverMetadata.issuer
+
+    return context.agent.resolveTrustChain({
+      entityIdentifier,
+      trustAnchors
+    })
+    .then((result) => !result || result.length === 0)
```

Comment:
What is result of resolveTrustChain? (I cannot see the type also) 
This looks like you return true when no trust chain was found, so returning true when not trusted. I may be wrong, but I can't see that clearly.

Reply:
fixed, it now only returns true if there are results in the array